include/bar.hpp

```diff
@@ -120,6 +128,10 @@ class Bar {
   std::unique_ptr<BarIpcClient> _ipc_client;
 #endif
   std::vector<std::shared_ptr<waybar::AModule>> modules_all_;
+
+  std::atomic<unsigned int> autohide_delay_ms = 0;
+  std::mutex autohide_mutex;
```

Comment:
All the UI rendering happens in a single thread. Why do you need this lock?

Reply:
Will remove it

Reply:
@alebastr , have removed it. Please review current corrected PR.